### PR TITLE
Run Functional Tests in CI

### PR DESCRIPTION
**Pull request by SharonHart**
Add additional job to CI pipeline for factional tests validation that:
1. Starts presidio cluster with docker-compose
2. Bootstraps the functional tests venv  
3. Runs the tests with pytest

### CODE REVIEW

File: azure-pipelines.yml
Comment (balteravishay) [inline azure-pipelines.yml:110]: Additional "set -eux" to bash tasks for logging and errors

File: azure-pipelines.yml
Comment (balteravishay) [inline azure-pipelines.yml:120]: Does this also output test results to ado?

**Review by balteravishay — COMMENTED** [review]
Some comments

File: azure-pipelines.yml
Comment (SharonHart) [reply azure-pipelines.yml:110]: Thanks missed that. exiting with nonzero code is not failing ado?

File: azure-pipelines.yml
Comment (SharonHart) [reply azure-pipelines.yml:120]: Yes, checked rhat

File: azure-pipelines.yml
Comment (SharonHart) [reply azure-pipelines.yml:111]: Done

**Review by navalev — CHANGES_REQUESTED** [review]
looks good. Lets move the functional tests to a seperate template to make sure the pipeline is readable

**Comment by SharonHart** [comment]
@navalev Will do, good idea

**Comment by SharonHart** [comment]
@navalev Done, approve once we merge the base and I'll rebase

**Review by navalev — APPROVED** [review]

File: presidio-anonymizer/Dockerfile
Comment (navalev) [inline presidio-anonymizer/Dockerfile:14]: since we are going to change this to flask run, please add a TODO comment with the relevant user id/bug

**Review by navalev — CHANGES_REQUESTED** [review]

File: presidio-anonymizer/Dockerfile
Comment (SharonHart) [reply presidio-anonymizer/Dockerfile]: Done

File: presidio-anonymizer/Dockerfile
Comment (shiranr) [inline presidio-anonymizer/Dockerfile:13]: A really small remark:
# [TODO ADO-2680] Use FLASK_RUN_PORT and flask run --host 0.0.0.0
Please add the TODO - this way it is being recognized as a TODO and marked accordingly in the code.

Comment (SharonHart) [reply presidio-anonymizer/Dockerfile:13]: missed that

File: .pipelines/templates/functional-tests.yml
Comment (shiranr) [inline .pipelines/templates/functional-tests.yml:21]: required dependencies maybe?

**Review by shiranr — APPROVED** [review]

**Review by shiranr — APPROVED** [review]

**Review by balteravishay — APPROVED** [review]

**Pull request merged**
